isos/appliance-extra.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+# Copyright 2018 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file should eventually be merged back into the main VIC appliance
+# build process. Common code between the regular appliance build and the
+# the extra-binary appliance build should be identified before merging.
+
+# Build the appliance filesystem ontop of the base
+
+# exit on failure and configure debug, include util functions
+set -e && [ -n "$DEBUG" ] && set -x
+DIR=$(dirname $(readlink -f "$0"))
+. $DIR/base/utils.sh
+
+
+function usage() {
+echo "Usage: $0 -p staged-package(tgz) -b binary-dir -x binary-source -f binary-filename (inside the ISO) -o appliance-output-name" 1>&2
+exit 1
+}
+
+while getopts "p:b:x:f:o:" flag
+do
+    case $flag in
+
+        p)
+            # Required. Package name
+            PACKAGE="$OPTARG"
+            ;;
+
+        b)
+            # Required. Target for iso and source for components
+            BIN="$OPTARG"
+            ;;
+
+        x)
+            # Required. Source of the extra binary to add to the ISO
+            EXTRABIN="$OPTARG"
+            ;;
+
+        f)
+            # Required. Filename of the extra binary inside the ISO
+            EXTRABIN_FILENAME="$OPTARG"
+            ;;
+
+        o)
+            # Required. Filename of the target appliance ISO
+            APPLIANCE_OUTNAME="$OPTARG"
+            ;;
+
+        *)
+            usage
+            ;;
+    esac
+done
+
+shift $((OPTIND-1))
+
+# check there were no extra args and the required ones are set
+if [ ! -z "$*" -o -z "$PACKAGE" -o -z "${BIN}" ]; then
+    usage
+fi
+
+if [ -z "${EXTRABIN}" ] || [ -z "${EXTRABIN_FILENAME}" ] || [ -z "${APPLIANCE_OUTNAME}" ]; then
```

Comment:
Is there a reason this is separate from the conditional above (and using a different style)? (Line 71)